src/combinatorics/derangements/basic.lean

```diff
@@ -24,105 +25,134 @@ We also define:
     sigma-type `Σ a : α, {f : perm α // only_possible_fixed_point f a}`.
 -/
 
-open equiv
-
-namespace derangements
-
-section definitions
+lemma function.mem_fixed_points_iff {α : Type*} {f : α → α} {x : α} :
+  x ∈ function.fixed_points f ↔ f x = x :=
+by refl
+
+lemma function.mem_fixed_points_apply {α : Type*} (f : α → α) {x : α}
+  (hx : x ∈ function.fixed_points f) :
+  f x ∈ function.fixed_points f :=
+by convert hx
+
+@[simp] lemma function.injective.mem_fixed_points_apply_iff {α : Type*} {f : α → α}
+  (hf : function.injective f) {x : α} :
+  f x ∈ function.fixed_points f ↔ x ∈ function.fixed_points f :=
+⟨λ h, hf h, function.mem_fixed_points_apply f⟩
+
+open equiv function
+
+/- Same as `(equiv.refl _)^.set.compl .symm.trans (subtype_equiv_right $ by simp)` but with better
+unfolding -/
+/-- Permutations on `sᶜ` are equivalent to permutations that fix `s` pointwise. -/
+protected def perm.compl_equiv {α : Type*} (s : set α) [decidable_pred (∈ s)] :
```

Comment:
is it generally better style to write these things as term-mode expressions then?